features/steps/attach.py

```diff
@@ -108,5 +109,22 @@ def when_i_attach_using_the_api_withot_enabling_services(context):
         '--data \'{{"token": "{}", "auto_enable_services": false}}\''
     )
     cmd = cmd.format(context.pro_config.contract_token)
-    print(cmd)
+    when_i_run_command(context, cmd, "with sudo")
+
+
+@when("I attach using the API by piping a file")
+def when_i_attach_using_the_api_by_piping_a_file(context):
+    when_i_create_file_with_content(
+        context,
+        file_path="/tmp/file.json",
+        text="""
+        {{
+          \"token\": \"{}\"
+        }}
+        """.format(
+            context.pro_config.contract_token
+        ),
+    )
+
+    cmd = "sh -c 'cat /tmp/file.json | pro api u.pro.attach.token.full_token_attach.v1 --data -'"  # noqa
     when_i_run_command(context, cmd, "with sudo")
```

Comment:
I suggest we place the emphasis on getting data from stdin, rather than piping a file, which is just one way of doing that. We also don't need to pipe a file for this test since we have support for setting stdin
```suggestion
@when("I attach using the API by writing token arg to stdin")
def when_i_attach_using_the_api_and_stdin(context):
    when_i_run_command(
        context,
        "pro api u.pro.attach.token.full_token_attach.v1 --data -",
        "with sudo",
        stdin=json.dumps({"token": context.pro_config.contract_token})
    )
```